Sources/DistributedActors/ActorSystem.swift

```diff
@@ -100,20 +100,37 @@ public final class ActorSystem: _Distributed.ActorTransport, @unchecked Sendable
     // ==== ----------------------------------------------------------------------------------------------------------------
     // MARK: Metrics
 
-    // TODO: Use ManagedAtomicLazyReference to store this
-    private lazy var _metrics: ActorSystemMetrics = ActorSystemMetrics(self.settings.metrics)
-    internal var metrics: ActorSystemMetrics {
-        self._metrics
-    }
+    internal let metrics: ActorSystemMetrics
 
     // ==== ----------------------------------------------------------------------------------------------------------------
     // MARK: Cluster
+    internal class Box<T> {
+        var value: T
+        init(_ value: T) {
+            self.value = value
+        }
+    }
 
     // initialized during startup
     // TODO: Use ManagedAtomicLazyReference to store this
     internal var _cluster: ClusterShell?
-    internal var _clusterControl: ClusterControl?
-    internal var _nodeDeathWatcher: NodeDeathWatcherShell.Ref?
+
+    private let _clusterControlStore: ManagedAtomicLazyReference<Box<ClusterControl>>
+    public var cluster: ClusterControl {
+        guard let box = _clusterControlStore.load() else {
+            fatalError("Somehow attempted to load system.cluster before it was initialized!")
+        }
+        return box.value
+    }
+
+    internal let _nodeDeathWatcherStore: ManagedAtomicLazyReference<Box<NodeDeathWatcherShell.Ref?>>
```

Comment:
I would have preferred to keep this private but I think it is needed for `_ActorShell.swift` since it needs to determine wether or not the property has been set (which it used to do using the `nil` value of the optional.

Reply:
It can check this based on if cluster is enabled or not I believe :) I'll give it a poke